src/zarr/abc/codec.py

```diff
@@ -138,7 +138,7 @@ def validate(
         """
 
     async def _decode_single(self, chunk_data: CodecOutput, chunk_spec: ArraySpec) -> CodecInput:
-        raise NotImplementedError
+        raise NotImplementedError  # pragma: no cover
```

Comment:
Out of curiosity, do you know why this isn't abstract if we're assuming that subclasses override it?

Reply:
I was wondering the same thing, I'm not sure why.

Reply:
Thanks, please don't go spelunking on my account :)